src/components/organism/Navigation.tsx

```diff
@@ -17,20 +21,60 @@ const links: Array<{ to: string; name: string }> = [
 const Navigation: React.FC<{ className?: string } & RouteComponentProps> = ({
   className,
   location: { pathname },
-}) => (
-  <nav className={className}>
-    <NavigationBrand />
-    <NavigationSeparator />
-    <div>
-      {links.map(l => (
-        <NavigationPill key={l.to} to={l.to} active={pathname === `/${l.to}`}>
-          {l.name}
-        </NavigationPill>
-      ))}
-    </div>
-    <NavigationSeparator style={{ marginTop: 'auto' }} />
-  </nav>
-);
+}) => {
+  const [auth, authDispatch] = useAuth();
+  const companies = useCompanyState();
+
+  const setActiveCompany = (id: string) => {
+    AuthActions.doSetActiveCompany(Number(id), authDispatch);
+  };
+
+  /**
+   * This is used for rendering and providing values to the select which allows the user
+   * to select a company.
+   */
+  const options = auth!.companies.map(c => {
+    const company = companies.find(e => e.id === c.company_id);
+
+    return {
+      // It might not be there yet, as we're loading this async in contexts/company
+      name: company
+        ? company.name
+        : `Loading company with ID ${c.company_id}...`,
+      value: c.company_id,
+    };
+  });
```

Comment:
I suggest that we make a spinning wheel  or some kind of loading effect when clicking on the dropdown list for selecting company, as long as not all companies are fetched.

I believe we induce less thinking (hence struggle) to the user if the application doesn't tell him/her specifically in the list of companies which are loaded and not.